Get basic engine working (with request spec)

knewter opened the pull request:
- implement combustion for 'dummy' app
- set up basic engine and root backend controller/view

This required moving from ABC -> Abc for the namespace because of conventions.
Basically, none of the controllers were being found successfully because rails
autoloading expects abc/foo_controller to define Abc::FooController.

Commit:
Get basic engine working (with request spec)

- implement combustion for 'dummy' app
- set up basic engine and root backend controller/view

This required moving from ABC -> Abc for the namespace because of conventions.
Basically, none of the controllers were being found successfully because rails
autoloading expects abc/foo_controller to define Abc::FooController.

parndt commented on abc.gemspec:
```diff
@@ -17,4 +17,6 @@ Gem::Specification.new do |s|
 
   s.required_ruby_version = '>= 1.9.3' # that's right.
+
+  s.add_dependency 'rails', '~>3.2.12'
```
Can you please just depend on the parts of Rails you're using? Rarely do we need the entire stack :heart:

parndt committed:
Updated all ABC to Abc

Conflicts:
	lib/abc.rb
	lib/abc/menu/menu.rb
	lib/abc/menu/menuable.rb
	spec/lib/menu/menu_spec.rb

parndt commented:
The tests got slow.

```
$ time ber spec/
.......

Finished in 0.19273 seconds
7 examples, 0 failures
Coverage report generated for RSpec to /code/parndt/afterburnercms/coverage. 27 / 27 LOC (100.0%) covered.

real    0m2.887s
user    0m2.564s
sys 0m0.311s
```

parndt committed:
Depending on Rails 4 and only parts of it.

knewter commented:
:heart_eyes: I've been dealing with babies so hadn't had time to see to all this. You guys are le nice

knewter commented:
@parndt @robyurkowski what's the consensus here?  Now that parndt's got this rails4-and-only-parts-of-it I think it's generally good to go.  Could be better vis-a-vis test speed I guess, but since nothing's happened in two days I'd rather see some forward momentum tbh.  Thoughts?

robyurkowski commented:
Okay, merging. Obviously there's some concern about test speed, but we'll just have to try to keep it as low as possible in coming months (sub-1s is my goal, personally). Loading rails is going to take a hammer to that number, and I don't know what a good solution is to that...
